build.cmd

```diff
@@ -227,19 +229,18 @@ if '%RestorePackages%' == 'true' (
 )
 
 
+rem Always do this it installs the dotnet cli and publishes the LKG
+rem ===============================================================
+
 set DOTNET_HOME=.\packages\dotnet
 set _dotnetexe=.\packages\dotnet\dotnet.exe
-if '%BUILD_CORECLR%' == '1' (
-  rem check to see if the dotnet cli tool exists
-  if not exist %_dotnetexe% (
-    echo Error: Could not find %_dotnetexe%.
-    rem do zipfile install nonsense
-    if not exist packages ( md packages )
-    if exist packages\dotnet ( rd packages /s /q )
-    powershell.exe -executionpolicy unrestricted -command .\scripts\install-dotnetcli.ps1 https://dotnetcli.blob.core.windows.net/dotnet/beta/Binaries/Latest/dotnet-dev-win-x64.latest.zip packages
-    @if ERRORLEVEL 1 echo Error: fetch dotnetcli failed && goto :failure
-    popd
-  )
+rem check to see if the dotnet cli tool exists
+if not exist %_dotnetexe% (
+  echo Could not find %_dotnetexe%. Do zipfile install
+  if not exist packages ( md packages )
+  if exist packages\dotnet ( rd packages\dotnet /s /q )
+  powershell.exe -executionpolicy unrestricted -command .\scripts\install-dotnetcli.ps1 https://dotnetcli.blob.core.windows.net/dotnet/beta/Binaries/Latest/dotnet-dev-win-x64.latest.zip packages
+  @if ERRORLEVEL 1 echo Error: fetch dotnetcli failed && goto :failure
```

Comment:
It’s nowhere near that big:
even fully expanded and building the coreclr the packages directory is only 1.25 GB.  Downloading the dotnet tools is necessary for us to go with a cross platform build, since it is going to be necessary to fetch and deploy the lkg.

If you run build without specifying coreclr the packages folder is 165MB
The dotnet tools download is a 33 mb zip.  The F# lkg is probably 4.5 MB.

The dotnet tools downloaded is only downloaded if it has not been already downloaded:

set DOTNET_HOME=.\packages\dotnet
set _dotnetexe=.\packages\dotnet\dotnet.exe
rem check to see if the dotnet cli tool exists
if not exist %_dotnetexe% (

I also have emotional problems with developers not using the coreclr version of the compiler.

The unfortunate problem with the current state of play is that when doing the coreclr will build the compiler 3 times:
              Proto
              Desktop
              Coreclr

Currently the LKG which runs on coreclr builds the Proto which runs on the desktop clr which builds the desktop and the coreclr compilers.

Shortly I expect to shrink that back down to:
Coreclr
Desktop

The LKG will build the coreclr version of the compiler, the coreclr version will build the desktop version, and all of the tests.

I have a branch where this mostly works, but I have to figure out what to do with msbuild and the buildtask cross platform.

Kevin

From: Don Syme [mailto:notifications@github.com]
Sent: Saturday, April 9, 2016 4:07 PM
To: Microsoft/visualfsharp <visualfsharp@noreply.github.com>
Cc: Kevin Ransom <Kevin.Ransom@microsoft.com>
Subject: Re: [Microsoft/visualfsharp] Fix nuget gen on multi-core machines, fix build issue when not building for coreclr (#1053)


In build.cmd<https://github.com/Microsoft/visualfsharp/pull/1053#discussion_r59123300>:

> -  if not exist %_dotnetexe% (

> -    echo Error: Could not find %_dotnetexe%.

> -    rem do zipfile install nonsense

> -    if not exist packages ( md packages )

> -    if exist packages\dotnet ( rd packages /s /q )

> -    powershell.exe -executionpolicy unrestricted -command .\scripts\install-dotnetcli.ps1 https://dotnetcli.blob.core.windows.net/dotnet/beta/Binaries/Latest/dotnet-dev-win-x64.latest.zip packages

> -    @if ERRORLEVEL 1 echo Error: fetch dotnetcli failed && goto :failure

> -    popd

> -  )

> +rem check to see if the dotnet cli tool exists

> +if not exist %_dotnetexe% (

> +  echo Could not find %_dotnetexe%. Do zipfile install

> +  if not exist packages ( md packages )

> +  if exist packages\dotnet ( rd packages\dotnet /s /q )

> +  powershell.exe -executionpolicy unrestricted -command .\scripts\install-dotnetcli.ps1 https://dotnetcli.blob.core.windows.net/dotnet/beta/Binaries/Latest/dotnet-dev-win-x64.latest.zip packages

> +  @if ERRORLEVEL 1 echo Error: fetch dotnetcli failed && goto :failure

This requires an download + installation of CoreCLR for all users of the repo, even when not doing anything to do with CoreCLR, e.g. when trying to fix bugs. Since CoreCLR is about 5GB (or seems to be, I stopped the download after 20 minutes after nearing 1GB) I don't think we can require it as a download in all development paths.

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub<https://github.com/Microsoft/visualfsharp/pull/1053/files/34bcaf8d4da4dd6d87b9f1855517b7514ca6fa4b#r59123300>